pkg/ipam/node_manager.go

```diff
@@ -273,6 +273,7 @@ func (n *NodeManager) GetNames() (allNodeNames []string) {
 // Kubernetes apiserver. The CiliumNode will be created if it didn't exist before.
 func (n *NodeManager) Upsert(resource *v2.CiliumNode) {
 	n.mutex.Lock()
+	defer n.mutex.Unlock()
```

Comment:
There are multiple calls to `n.mutex.Unlock()` in this function that now need to be removed

Reply:
thanks good catch 😅